<u> **BACKGROUND** </u>

Aging signatures are changes in gene expression which accompany the biological process of aging. The goal of the Molecular Transducers of Physical Activity Consortium (MoTrPAC) is to assess molecular changes that occur in response to physical activity [1]. To investigate the common biological underpinnings that accompany both aging and exercise and discover genes that are induced and repressed due to exercise and also change in levels due to aging, we crossed the aging signatures created from GTEx [2] with the MoTrPAC rat endurance training [3] gene sets. 

<br />

<u> **METHOD** </u>

GTEx data provides 1000 samples of gene expression data ​​across 43 tissues from 175 individuals in order to examine how genetic expression varies among tissues [4]. To create GTEx aging signatures, for each tissue, we performed a differential expression analysis using limma voom that compares the expression levels of the young population (ages  20-29) with sex matched older populations e.g patients aged 20-29 vs 60-69.  Each generated gene set in the gene set library consisted of up/down differentially expressed genes for each tissue and younger-older comparison group. 

<br />

To create the MoTrPAC rat endurance gene sets, we performed a differential expression analysis which compared the gene expression levels of a training group vs sex-matched controls for each tissue at every time point (exercise duration) in the study. This data comes from the RNA-seq data created by the MoTrPAC Data Coordination Center (DCC) that measured the gene expression levels or rats across a range of ages and fitness levels by molecular probing of multiple tissues before and after acute and chronic exercise. The resulting dataframe was filtered for genes that have an adjusted p-value < 0.05 to get the gene expression signatures for specific tissue, sex, and time point group . We obtained the up and down differentially expressed genes for each specific tissue, sex, and time point group with up genes being genes with a log fold change (logFC) > 0 and down genes being genes with a log fold change (logFC) < 0. Each gene set in the generated gene set library consisted of up/down differentially expressed genes for each specific tissue, sex, and time point group. 

<br />

The CFDE GMT crossing feature allows users to cross the GMT files generated by Common Fund DCCs in order to find gene set pairs from the different GMTs that have a significant overlap. The Fisher's exact test is used to quantify the significance of such overlap. We crossed all pairs of gene sets across all libraries and retained gene sets pairs with a p-value < 0.001. The top 5000 pairs of sets of each crossing are saved and displayed in an interactive table. The overlapping genes between each pair can be sent to Enrichr [5] for enrichment analysis or to a GPT-4 model [6] for hypothesis generation about the potential reason for overlap between the gene sets that make up the crossed gene set pair. To create the GPT-4 generated hypothesis, we give GPT-4 a textual description of each gene set in the crossed pair, the overlapping genes, the enriched terms from Enrichr using the GO Biological Processes [7], WikiPathway 2023 Human [8], MGI Mammalian Phenotype Level 4 [9], and GWAS Catalog 2023 [10] libraries as backgrounds, and then ask it to compose an abstract that describes the potential connection between these gene sets. 

<br />

<u> **RESULT AND DISCUSSION** </u>

The top two crossing gene set pairs of the GTEx Aging Signatures and MoTrPAC Rat Endurance Training GMTs (GTEx Blood 20-29 vs 60-69 Up ∩ T30-Blood-Rna Female 2W Down and GTEx Blood 20-29 vs 70-79 Up ∩ T30-Blood-Rna Female 2W Down) have 35 (p-value=6.52e-38) and 26 (p-value=1.05e-24) overlapping genes respectively. We also find that these sets share 24 genes in common (Fig 1.). The significant overlap between the top crossings further suggests the presence of shared biological pathways influenced by the aging of tissue and exercise. 

<br />

![Venn Diagram Top 2 Aging Signatures vs MoTrPAC Crossing](/img/markdownImg/gtex_motrpac_venn.png)
<h5> Fig. 1: Venn diagram showing the overlap between A:  GTEx Blood 20-29 vs 70-79 Up ∩ T30-Blood-Rna Female 2W Down and B: GTEx Blood 20-29 vs 60-69 Up ∩ T30-Blood-Rna Female 2W Down gene sets. </h5>
<br />

In performing enrichment analysis on the overlapping genes of these pairs by sending them to Enrichr, we find that the overlapping genes are enriched for pathways are related to immune response, blood coagulation, and lipid metabolism (Fig. 2), which are all processes that can be affected by both aging and physical activity. 

<br />

![Enriched Terms from KEGG 2021 Human Library of Overlapping Genes of Top 2 Crossed Pairs](/img/markdownImg/KEGG_2021_Human_bar_graph.svg)
<h5> Fig. 2: Bar plot showing enriched terms from the KEGG 2021 Human library based on the  overlapping genes between A:  GTEx Blood 20-29 vs 70-79 Up ∩ T30-Blood-Rna Female 2W Down and B: GTEx Blood 20-29 vs 60-69 Up ∩ T30-Blood-Rna Female 2W Down gene sets.  </h5>
<br />


The GPT-4 generated hypothesis further explains the possible ways that these overlapping genes are tied to the physiological changes that accompany both aging and exercise through these enriched pathways. It proposes that the high overlap between the two gene sets could be due to the shared influence of aging and physical activity on these biological pathways and processes in opposing ways. The GPT-4 generated hypothesis posits that the overlap is because “Aging is associated with changes in lipid metabolism and increased risk of cardiovascular diseases, while regular physical activity can improve lipid profile and reduce the risk of cardiovascular diseases. Aging can lead to changes in blood coagulation, and physical activity is known to influence blood viscosity and coagulation.”

<br />

![Novelty Assessment Top 2 Aging Signatures vs MoTrPAC Crossing](/img/markdownImg/gtex_motrpac_crossing.png)
<h5> Fig. 3: Bar plot showing how many publications mention each gene in  A:  GTEx Blood 20-29 vs 70-79 Up ∩ T30-Blood-Rna Female 2W Down and B: GTEx Blood 20-29 vs 60-69 Up ∩ T30-Blood-Rna Female 2W Down gene sets color coded by appearance of gene in only one set or in both sets. </h5>
<br />

<u> **CONCLUSION** </u>

Here, we found some biological pathways and their related genes that are associated with both aging and exercise through crossing aging signatures created from GTEx and exercise related gene sets from MoTrPAC. Many of these genes are mentioned in less than 100 publications which suggests that they might be understudied. This provides evidence for further exploration of the link between these genes and both aging and exercise (Fig.3). This displays the utility of the G2SG application in integrating data sets produced from various Common Fund programs to stimulate scientific discovery.

<br />

<u> **REFERENCES** </u>

- [1]	J. A. Sanford et al., “Molecular Transducers of Physical Activity Consortium (MoTrPAC): Mapping the Dynamic Responses to Exercise,” Cell, vol. 181, no. 7, pp. 1464–1474, Jun. 2020.

- [2]	K. Jia, C. Cui, Y. Gao, Y. Zhou, and Q. Cui, “An analysis of aging-related genes derived from the Genotype-Tissue Expression project (GTEx),” Cell Death Discov, vol. 4, p. 26, Aug. 2018.

- [3]	S. Schenk et al., “Physiological Adaptations to Progressive Endurance Exercise Training in Adult And Aged Rats: Insights from The Molecular Transducers of Physical Activity Consortium (MoTrPAC),” Function, doi: 10.1093/function/zqae014.

- [4]	T. G. Consortium et al., “The Genotype-Tissue Expression (GTEx) pilot analysis: Multitissue gene regulation in humans,” Science, vol. 348, no. 6235, pp. 648–660, 2015.

- [5]	E. Y. Chen et al., “Enrichr: interactive and collaborative HTML5 gene list enrichment analysis tool,” BMC Bioinformatics, vol. 14, p. 128, Apr. 2013.

- [6]	OpenAI et al., “GPT-4 Technical Report,” arXiv [cs.CL], Mar. 15, 2023. [Online]. Available: http://arxiv.org/abs/2303.08774

- [7]	Gene Ontology Consortium, “Gene Ontology Consortium: going forward,” Nucleic Acids Res., vol. 43, no. Database issue, pp. D1049–56, Jan. 2015.

- [8]	M. Kutmon et al., “WikiPathways: capturing the full diversity of pathway knowledge,” Nucleic Acids Res., vol. 44, no. D1, pp. D488–94, Jan. 2016.

- [9]	J. A. Blake, C. J. Bult, J. T. Eppig, J. A. Kadin, J. E. Richardson, and Mouse Genome Database Group, “The Mouse Genome Database 
genotypes::phenotypes,” Nucleic Acids Res., vol. 37, no. Database issue, pp. D712–9, Jan. 2009.

[10]	E. Sollis et al., “The NHGRI-EBI GWAS Catalog: knowledgebase and deposition resource,” Nucleic Acids Res., vol. 51, no. D1, pp. D977–D985, Jan. 2023.