<u> **BACKGROUND** </u>

Aging is the most profound risk factor for many chronic and non-infectious diseases such as diabetes [1], cardiovascular disease [2] and neurological diseases such as Alzheimer's [3] and Parkinson’s disease [4]. This risk has been linked to the overlap between the molecular basis of these diseases and that of aging [5], [6]. Moderate exercise (planned and intentional physical activity [7]) has become widely accepted as to promote health and aid the prevention of diseases [8] with exercise being shown to induce organelle, cellular, blood-brain and external barrier protection [36], [37], protein proteostasis [9], DNA repair [10], anti-inflammation (as C-reactive protein concentration is strongly related to physical activity [11]), autophagy regulation [12], and other processes which positively contribute to better health and many of which counter the molecular mechanisms of aging [13]. 
Aging signatures are changes in gene expression which accompany the biological process of aging. The goal of the Molecular Transducers of Physical Activity Consortium (MoTrPAC) is to assess molecular changes that occur in response to physical activity [14]. To investigate the common biological underpinnings that accompany both aging and exercise and discover genes that are induced and repressed due to exercise and also change in levels due to aging, we crossed the aging signatures created from GTEx [15] with the MoTrPAC rat endurance training [16] gene sets. 

<br />

<u> **METHOD** </u>

GTEx data provides 1000 samples of gene expression data ​​across 43 tissues from 175 individuals in order to examine how genetic expression varies among tissues [17]. To create GTEx aging signatures, for each tissue, we performed a differential expression analysis using limma voom that compares the expression levels of the young population (ages  20-29) with sex matched older populations e.g patients aged 20-29 vs 60-69.  Each generated gene set in the gene set library consisted of up/down differentially expressed genes for each tissue and younger-older comparison group. 

<br />

To create the MoTrPAC rat endurance gene sets, we performed a differential expression analysis which compared the gene expression levels of a training group vs sex-matched controls for each tissue at every time point (exercise duration) in the study. This data comes from the RNA-seq data created by the MoTrPAC Data Coordination Center (DCC) that measured the gene expression levels or rats across a range of ages and fitness levels by molecular probing of multiple tissues before and after acute and chronic exercise. The resulting dataframe was filtered for genes that have an adjusted p-value < 0.05 to get the gene expression signatures for specific tissue, sex, and time point group . We obtained the up and down differentially expressed genes for each specific tissue, sex, and time point group with up genes being genes with a log fold change (logFC) > 0 and down genes being genes with a log fold change (logFC) < 0. Each gene set in the generated gene set library consisted of up/down differentially expressed genes for each specific tissue, sex, and time point group. 

<br />

The CFDE GMT crossing feature allows users to cross the GMT files generated by Common Fund DCCs in order to find gene set pairs from the different GMTs that have a significant overlap. The Fisher's exact test is used to quantify the significance of such overlap. We crossed all pairs of gene sets across all libraries and retained gene sets pairs with a p-value < 0.001. The top 5000 pairs of sets of each crossing are saved and displayed in an interactive table. The overlapping genes between each pair can be sent to Enrichr [18] for enrichment analysis or to a GPT-4 model [19] for hypothesis generation about the potential reason for overlap between the gene sets that make up the crossed gene set pair. To create the GPT-4 generated hypothesis, we give GPT-4 a textual description of each gene set in the crossed pair, the overlapping genes, the enriched terms from Enrichr using the GO Biological Processes [20], WikiPathway 2023 Human [21], MGI Mammalian Phenotype Level 4 [22], and GWAS Catalog 2023 [23] libraries as backgrounds, and then ask it to compose an abstract that describes the potential connection between these gene sets. 

<br />

<u> **RESULT AND DISCUSSION** </u>

Examination of the most significantly overlapping terms from the GTEx aging signatures crossed with the MoTrPAC rat endurance training reveals several pairs of gene sets with high overlap from adipose tissue and adrenal gland. Specifically, the gene set pairs “GTEx AdiposeTissue 20-29 vs 50-59 Down ∩ T60-Adrenal Male 1W Up”, “GTEx AdiposeTissue 20-29 vs 60-69 Down ∩ T60-Adrenal Male 1W Up”, and “GTEx AdiposeTissue 20-29 vs 60-69 Down ∩ T60-Adrenal Male 1W Up” are in the top 20 most overlapping gene set pairs. Comparing these sets utilizing the Combine module in GeneSetCart reveals 14 shared genes across the three set pairs with a total of 36 unique genes. The “GTEx AdiposeTissue 20-29 vs 50-59 Down” gene set contains genes downregulated in adipose tissue from samples taken from 50-59-year-olds compared to those from 20-29-year-olds whereas “T60-Adrenal Male 1W Up” contains genes upregulated in male rat adrenal tissue following 1 week of exercise. Creating GPT-4 hypothesis for the crossed sets GTEx AdiposeTissue 20-29 vs 50-59 Down ∩ T60-Adrenal Male 1W Up provides several shared mechanisms:


<br />

![Enrichment Results from Adrenal and Adipose Overlap](/img/markdownImg/adrenal_adipose_fig.png)
<h5>Figure 1. Exploring shared pathways implicated in aging and exercise in adipose tissue and adrenal cortex. A. Intersecting genes between three aging GTEx signatures from adipose tissue and exercise from adrenal cortex. B. Top enriched terms from WikiPathway 2023 Human library of intersecting gene signature. C. Top enriched terms from GO Biological Processes library of intersecting gene signature D. Top enriched terms from MGI Mammalian Phenotypes Level 4 library of intersecting gene signature.</h5>
<br />

The high overlap between the two gene sets can be attributed to the shared biological pathways and processes that are influenced by both aging in AdiposeTissue and the molecular changes induced by exercise in the Adrenal in Male subjects. The enrichment analysis reveals that the overlapping genes are significantly involved in pathways such as Glycerolipids And Glycerophospholipids WP4722, Triacylglyceride Synthesis WP325, and Adipogenesis WP236, which are crucial for lipid metabolism and adipocyte function. These pathways are essential for maintaining energy homeostasis and are likely to be affected by both aging and exercise. For instance, genes like PFKFB3 and PSAT1 are involved in the Triglyceride Biosynthetic Process (GO:0019432) and Fatty Acid Metabolic Process (GO:0006631), which are critical for energy storage and utilization. Additionally, the regulation of cold-induced thermogenesis (GO:0120161) and positive regulation of cold-induced thermogenesis (GO:0120162) are processes that are vital for maintaining body temperature and metabolic rate, both of which can be influenced by aging and physical activity. The presence of genes such as ADRB3 and G0S2 in these pathways suggests a role in the differentiation of white and brown adipocytes (WP2895) and enhanced lipolysis (MP:0008034), which are important for adapting to metabolic changes. Furthermore, the involvement of genes like FASN and DGAT2 in the Percent Liver Fat and decreased liver triglyceride level (MP:0009356) pathways indicates a connection to liver health and lipid metabolism, which are affected by both aging and exercise. The overlap in genes related to the Positive Regulation Of Metabolic Process (GO:0009893) and abnormal white adipose tissue morphology (MP:0002970) further supports the idea that these gene sets are linked through their roles in metabolic regulation and adipose tissue function. Overall, the enrichment analysis highlights the interconnected nature of these biological processes, explaining the significant overlap between the gene sets.” (Fig. 1A-C)

<br />


In an analysis of transcription factors most likely to regulate the overlapping genes between the three sets using ChEA3 [1], PPARG is the top ranked with 10 out of 14 of the genes being known targets of PPARG (Fig. 1D). PPARG is a key regulator of lipid metabolism, and adipogenesis, and a known therapeutic target for type 2 diabetes [2]. A related TF in the same family, PPARD, which is not included in ChEA, promotes cardiovascular endurance through the conservation of glucose [3]. Agonists of PPARD, such as GW501516 also known as Endurobol, have been used by elite athletes to improve their performance and are banned by the World Anti-Doping agency. This crossing and the identified shared genes may be key understudied genes in the modulation of response to exercise as it relates to aging.

<br />

<u> **REFERENCES** </u>

- [1]	H. L. C. Wilkerson, “Problems of an Aging Population,” Am. J. Public Health Nations. Health, vol. 37, no. 2, pp. 177–188, Feb. 1947.

- [2]	B. J. North and D. A. Sinclair, “The intersection between aging and cardiovascular disease,” Circ. Res., vol. 110, no. 8, pp. 1097–1108, Apr. 2012.

- [3]	X. Xia, Q. Jiang, J. McDermott, and J.-D. J. Han, “Aging and Alzheimer’s disease: Comparison and associations from molecular to system level,” Aging Cell, vol. 17, no. 5, p. e12802, Oct. 2018.

- [4]	A. Reeve, E. Simcox, and D. Turnbull, “Ageing and Parkinson’s disease: why is advancing age the biggest risk factor?,” Ageing Res. Rev., vol. 14, no. 100, pp. 19–30, Mar. 2014.

- [5]	G. Wick, P. Jansen-Dürr, P. Berger, I. Blasko, and B. Grubeck-Loebenstein, “Diseases of aging,” Vaccine, vol. 18, no. 16, pp. 1567–1583, Feb. 2000.

- [6]	D. Saul and R. L. Kosinsky, “Epigenetics of Aging and Aging-Associated Diseases,” Int. J. Mol. Sci., vol. 22, no. 1, Jan. 2021, doi: 10.3390/ijms22010401.

- [7]	C. J. Caspersen, K. E. Powell, and G. M. Christenson, “Physical activity, exercise, and physical fitness: definitions and distinctions for health-related research,” Public Health Rep., vol. 100, no. 2, pp. 126–131, Mar-Apr 1985.

- [8]	C. Fiuza-Luces et al., “Exercise benefits in cardiovascular disease: beyond attenuation of traditional risk factors,” Nat. Rev. Cardiol., vol. 15, no. 12, pp. 731–743, Dec. 2018.

- [9]	M. A. Małkiewicz, A. Szarmach, A. Sabisz, W. J. Cubała, E. Szurowska, and P. J. Winklewski, “Blood-brain barrier permeability and physical exercise,” J. Neuroinflammation, vol. 16, no. 1, p. 15, Jan. 2019.

- [10]	P. S. Souza et al., “Physical Exercise Attenuates Experimental Autoimmune Encephalomyelitis by Inhibiting Peripheral Immune Response and Blood-Brain Barrier Disruption,” Mol. Neurobiol., vol. 54, no. 6, pp. 4723–4737, Aug. 2017.

- [11]	R. V. Musci, K. L. Hamilton, and B. F. Miller, “Targeting mitochondrial function and proteostasis to mitigate dynapenia,” Eur. J. Appl. Physiol., vol. 118, no. 1, pp. 1–9, Jan. 2018.

- [12]	Z. Radák et al., “Exercise training decreases DNA damage and increases DNA repair and resistance against oxidative stress of proteins in aged rat skeletal muscle,” Pflugers Arch., vol. 445, no. 2, pp. 273–278, Nov. 2002.

- [13]	E. S. Ford, “Does exercise reduce inflammation? Physical activity and C-reactive protein among U.S. adults,” Epidemiology, vol. 13, no. 5, pp. 561–568, Sep. 2002.

- [14] J. A. Sanford et al., “Molecular Transducers of Physical Activity Consortium (MoTrPAC): Mapping the Dynamic Responses to Exercise,” Cell, vol. 181, no. 7, pp. 1464–1474, Jun. 2020.

- [15] K. Jia, C. Cui, Y. Gao, Y. Zhou, and Q. Cui, “An analysis of aging-related genes derived from the Genotype-Tissue Expression project (GTEx),” Cell Death Discov, vol. 4, p. 26, Aug. 2018.

- [16] S. Schenk et al., “Physiological Adaptations to Progressive Endurance Exercise Training in Adult And Aged Rats: Insights from The Molecular Transducers of Physical Activity Consortium (MoTrPAC),” Function, doi: 10.1093/function/zqae014.

- [17] T. G. Consortium et al., “The Genotype-Tissue Expression (GTEx) pilot analysis: Multitissue gene regulation in humans,” Science, vol. 348, no. 6235, pp. 648–660, 2015.

- [18] E. Y. Chen et al., “Enrichr: interactive and collaborative HTML5 gene list enrichment analysis tool,” BMC Bioinformatics, vol. 14, p. 128, Apr. 2013.

- [19] OpenAI et al., “GPT-4 Technical Report,” arXiv [cs.CL], Mar. 15, 2023. [Online]. Available: http://arxiv.org/abs/2303.08774

- [20] Gene Ontology Consortium, “Gene Ontology Consortium: going forward,” Nucleic Acids Res., vol. 43, no. Database issue, pp. D1049–56, Jan. 2015.

- [21] M. Kutmon et al., “WikiPathways: capturing the full diversity of pathway knowledge,” Nucleic Acids Res., vol. 44, no. D1, pp. D488–94, Jan. 2016.

- [22] J. A. Blake, C. J. Bult, J. T. Eppig, J. A. Kadin, J. E. Richardson, and Mouse Genome Database Group, “The Mouse Genome Database genotypes::phenotypes,” Nucleic Acids Res., vol. 37, no. Database issue, pp. D712–9, Jan. 2009.

- [23] E. Sollis et al., “The NHGRI-EBI GWAS Catalog: knowledgebase and deposition resource,” Nucleic Acids Res., vol. 51, no. D1, pp. D977–D985, Jan. 2023.

- [24] A. B. Keenan et al., “ChEA3: transcription factor enrichment analysis by orthogonal omics integration,” Nucleic Acids Res., vol. 47, no. W1, pp. W212–W224, Jul. 2019.

- [25]	M. Ahmadian et al., “PPARγ signaling and metabolism: the good, the bad and the future,” Nat. Med., vol. 19, no. 5, pp. 557–566, May 2013.

- [26]	W. Fan et al., “PPARδ promotes running endurance by preserving glucose,” Cell Metab., vol. 25, no. 5, pp. 1186–1193.e4, May 2017.
