<u> **BACKGROUND** </u>

Aging is the most profound risk factor for many chronic and non-infectious diseases such as diabetes [1], cardiovascular disease [2] and neurological diseases such as Alzheimer's [3] and Parkinson’s disease [4]. This risk has been linked to the overlap between the molecular basis of these diseases and that of aging [5], [6]. Moderate exercise (planned and intentional physical activity [7]) has become widely accepted as to promote health and aid the prevention of diseases [8] with exercise being shown to induce organelle, cellular, blood-brain and external barrier protection [36], [37], protein proteostasis [9], DNA repair [10], anti-inflammation (as C-reactive protein concentration is strongly related to physical activity [11]), autophagy regulation [12], and other processes which positively contribute to better health and many of which counter the molecular mechanisms of aging [13]. 
Aging signatures are changes in gene expression which accompany the biological process of aging. The goal of the Molecular Transducers of Physical Activity Consortium (MoTrPAC) is to assess molecular changes that occur in response to physical activity [14]. To investigate the common biological underpinnings that accompany both aging and exercise and discover genes that are induced and repressed due to exercise and also change in levels due to aging, we crossed the aging signatures created from GTEx [15] with the MoTrPAC rat endurance training [16] gene sets. 

<br />

![Aging Signatures - MoTrPAC GMT Crossing Workflow](/img/markdownImg/gtex_motrpac_workflow.png)
<h5> Fig. 1: Exploring shared pathways implicated in aging and exercise use case workflow. </h5>
<br />

<u> **METHOD** </u>

GTEx data provides 1000 samples of gene expression data ​​across 43 tissues from 175 individuals in order to examine how genetic expression varies among tissues [17]. To create GTEx aging signatures, for each tissue, we performed a differential expression analysis using limma voom that compares the expression levels of the young population (ages  20-29) with sex matched older populations e.g patients aged 20-29 vs 60-69.  Each generated gene set in the gene set library consisted of up/down differentially expressed genes for each tissue and younger-older comparison group. 

<br />

To create the MoTrPAC rat endurance gene sets, we performed a differential expression analysis which compared the gene expression levels of a training group vs sex-matched controls for each tissue at every time point (exercise duration) in the study. This data comes from the RNA-seq data created by the MoTrPAC Data Coordination Center (DCC) that measured the gene expression levels or rats across a range of ages and fitness levels by molecular probing of multiple tissues before and after acute and chronic exercise. The resulting dataframe was filtered for genes that have an adjusted p-value < 0.05 to get the gene expression signatures for specific tissue, sex, and time point group . We obtained the up and down differentially expressed genes for each specific tissue, sex, and time point group with up genes being genes with a log fold change (logFC) > 0 and down genes being genes with a log fold change (logFC) < 0. Each gene set in the generated gene set library consisted of up/down differentially expressed genes for each specific tissue, sex, and time point group. 

<br />

The CFDE GMT crossing feature allows users to cross the GMT files generated by Common Fund DCCs in order to find gene set pairs from the different GMTs that have a significant overlap. The Fisher's exact test is used to quantify the significance of such overlap. We crossed all pairs of gene sets across all libraries and retained gene sets pairs with a p-value < 0.001. The top 5000 pairs of sets of each crossing are saved and displayed in an interactive table. The overlapping genes between each pair can be sent to Enrichr [18] for enrichment analysis or to a GPT-4 model [19] for hypothesis generation about the potential reason for overlap between the gene sets that make up the crossed gene set pair. To create the GPT-4 generated hypothesis, we give GPT-4 a textual description of each gene set in the crossed pair, the overlapping genes, the enriched terms from Enrichr using the GO Biological Processes [20], WikiPathway 2023 Human [21], MGI Mammalian Phenotype Level 4 [22], and GWAS Catalog 2023 [23] libraries as backgrounds, and then ask it to compose an abstract that describes the potential connection between these gene sets. 

<br />

<u> **RESULT AND DISCUSSION** </u>

Crossing the GTEx Aging Signatures and MoTrPAC Rat Endurance Training GMTs yields 346 gene set pairs with significant overlap (p-value < 0.001). The top two crossing gene set pairs (GTEx Blood 20-29 vs 60-69 Up ∩ T30-Blood-Rna Female 2W Down and GTEx Blood 20-29 vs 70-79 Up ∩ T30-Blood-Rna Female 2W Down) have 35 (p-value=6.52e-38) and 26 (p-value=1.05e-24) overlapping genes respectively. Furthermore, by adding these overlapping genes to our cart and using them to start an interactive session, the intersection set operation of the top two pairs shows these sets share 24 genes in common

<br />

![Venn Diagram Top 2 Aging Signatures vs MoTrPAC Crossing](/img/markdownImg/gtex_motrpac_venn.png)
<h5> Fig. 2: Overlap between top 2 crossing results of GTEx Aging Signatures vs MoTrPAC Rat Endurance Training libraries. </h5>
<br />

The GTEx Blood 20-29 vs 60-69 Up gene set contains genes that are upregulated when comparing the blood of subjects aged 60-60 to those aged 20-29. GTEx Blood 20-29 vs 70-79 Up gene set contains genes that are upregulated when comparing the blood of subjects aged 60-60 to those aged 20-29. The T30-Blood-Rna Female 2W Down gene set consists of genes that are downregulated in the blood of rats after four weeks of endurance training. The enrichment analysis found enriched pathways related to immune response, blood coagulation, and lipid metabolism, which are all processes that are well accepted to be affected by aging and physical activity (Fig. 9). Some enriched terms are blood related processes that are particularly known to undergo significant changes with aging and exercise such as blood coagulation and fibrinolysis. It is well known that aging is associated with increased plasma levels of many proteins of blood coagulation [24]. Additionally, short term exercise is also associated with transient increase in blood coagulation, moderate exercise with enhancing blood fibrinolytic activity without activation of blood coagulation mechanisms while heavy exercise induces simultaneous activation of blood fibrinolysis and coagulation [25]. Research also shows that blood lipids are a likely source of human aging biomarkers with blood lipid levels (including total cholesterol, low- and high-density lipoprotein cholesterol, and triglyceride concentrations) changing in specific ways with age [26], [27] while endurance exercise induces fat oxidation [28].

<br />


![Enriched Terms from KEGG 2021 Human Library of Overlapping Genes of Top 2 Crossed Pairs](/img/markdownImg/gtex_motrpac_enrichment.png)
<h5> Fig. 3: Enrichment analysis of top overlapping genes of the top 2 crossing results of GTEx Aging Signatures vs MoTrPAC Rat Endurance Training libraries. (A) Enriched terms from the KEGG 2021 Human library based on the overlapping genes between A:  GTEx Blood 20-29 vs 70-79 Up ∩ T30-Blood-Rna Female 2W Down and B: GTEx Blood 20-29 vs 60-69 Up ∩ T30-Blood-Rna Female 2W Down gene sets. Link: https://maayanlab.cloud/Enrichr/enrich?dataset=d1a0862c3a236d626ef4567f96f4b526. (B) Enriched terms from the GO Biological Processes library based on the overlapping genes between A:  GTEx Blood 20-29 vs 70-79 Up ∩ T30-Blood-Rna Female 2W Down and B: GTEx Blood 20-29 vs 60-69 Up ∩ T30-Blood-Rna Female 2W Down gene sets. Link: https://maayanlab.cloud/Enrichr/enrich?dataset=d1a0862c3a236d626ef4567f96f4b526 
  </h5>
<br />


The GPT-4 generated hypothesis further explains the possible ways that these overlapping genes are tied to the physiological changes that accompany both aging and exercise through these enriched pathways. It proposes that the high overlap between the two gene sets could be due to the shared influence of aging and physical activity on these biological pathways and processes in opposing ways. The GPT-4 generated hypothesis posits that the overlap is because “Aging is associated with changes in lipid metabolism and increased risk of cardiovascular diseases, while regular physical activity can improve lipid profile and reduce the risk of cardiovascular diseases. Aging can lead to changes in blood coagulation, and physical activity is known to influence blood viscosity and coagulation.”

<br />


This is in line with studies which have shown that exercise can reverse many of the hallmarks of aging through anti-aging mechanisms [31] giving it therapeutic potential for aging related diseases. 

<br />

A novelty assessment of the top crossing pair sets was done by comparing the number of publications each gene is associated with according to GeneRIF (Fig. 10). We find that 65% of the genes contained in those sets are associated in less than 100 publications in PubMed with two genes (HAO1 and SLC25A47) being associated with less than 10 publications. We also find that of these 37 genes, only 11 are found to be related to aging or exercise (CRP, HNF4A, TTR, HAMP, FGB, RBP4, FGA, APOC1, HABP2, SERPINA4 and AGXT2). These results suggest that many of our observed genes are understudied in the context of both aging and exercise, and the relationship between these two processes that our crossing results have shown to be significantly related. This provides evidence for the need for further exploration of the link between these genes and both aging and exercise. 

<br />

![Novelty Assessment Top 2 Aging Signatures vs MoTrPAC Crossing](/img/markdownImg/gtex_motrpac_novelty.png)
<h5> Fig. 4: Novelty assessment of top 2 crossing results. (A): Genes grouped by which gene set they appeared in. (B): Showing if publications genes are associated with are related to aging, exercise,or both terms based on GeneRIF </h5>
<br />

<u> **CONCLUSION** </u>

Here, we found some biological pathways and their related genes that are associated with both aging and exercise through crossing aging signatures created from GTEx and exercise related gene sets from MoTrPAC. Many of these genes are mentioned in less than 100 publications which suggests that they might be understudied. This provides evidence for further exploration of the link between these genes and both aging and exercise (Fig.3). This displays the utility of the G2SG application in integrating data sets produced from various Common Fund programs to stimulate scientific discovery.

<br />

<u> **REFERENCES** </u>

- [1]	H. L. C. Wilkerson, “Problems of an Aging Population,” Am. J. Public Health Nations. Health, vol. 37, no. 2, pp. 177–188, Feb. 1947.

- [2]	B. J. North and D. A. Sinclair, “The intersection between aging and cardiovascular disease,” Circ. Res., vol. 110, no. 8, pp. 1097–1108, Apr. 2012.

- [3]	X. Xia, Q. Jiang, J. McDermott, and J.-D. J. Han, “Aging and Alzheimer’s disease: Comparison and associations from molecular to system level,” Aging Cell, vol. 17, no. 5, p. e12802, Oct. 2018.

- [4]	A. Reeve, E. Simcox, and D. Turnbull, “Ageing and Parkinson’s disease: why is advancing age the biggest risk factor?,” Ageing Res. Rev., vol. 14, no. 100, pp. 19–30, Mar. 2014.

- [5]	G. Wick, P. Jansen-Dürr, P. Berger, I. Blasko, and B. Grubeck-Loebenstein, “Diseases of aging,” Vaccine, vol. 18, no. 16, pp. 1567–1583, Feb. 2000.

- [6]	D. Saul and R. L. Kosinsky, “Epigenetics of Aging and Aging-Associated Diseases,” Int. J. Mol. Sci., vol. 22, no. 1, Jan. 2021, doi: 10.3390/ijms22010401.

- [7]	C. J. Caspersen, K. E. Powell, and G. M. Christenson, “Physical activity, exercise, and physical fitness: definitions and distinctions for health-related research,” Public Health Rep., vol. 100, no. 2, pp. 126–131, Mar-Apr 1985.

- [8]	C. Fiuza-Luces et al., “Exercise benefits in cardiovascular disease: beyond attenuation of traditional risk factors,” Nat. Rev. Cardiol., vol. 15, no. 12, pp. 731–743, Dec. 2018.

- [9]	M. A. Małkiewicz, A. Szarmach, A. Sabisz, W. J. Cubała, E. Szurowska, and P. J. Winklewski, “Blood-brain barrier permeability and physical exercise,” J. Neuroinflammation, vol. 16, no. 1, p. 15, Jan. 2019.

- [10]	P. S. Souza et al., “Physical Exercise Attenuates Experimental Autoimmune Encephalomyelitis by Inhibiting Peripheral Immune Response and Blood-Brain Barrier Disruption,” Mol. Neurobiol., vol. 54, no. 6, pp. 4723–4737, Aug. 2017.

- [11]	R. V. Musci, K. L. Hamilton, and B. F. Miller, “Targeting mitochondrial function and proteostasis to mitigate dynapenia,” Eur. J. Appl. Physiol., vol. 118, no. 1, pp. 1–9, Jan. 2018.

- [12]	Z. Radák et al., “Exercise training decreases DNA damage and increases DNA repair and resistance against oxidative stress of proteins in aged rat skeletal muscle,” Pflugers Arch., vol. 445, no. 2, pp. 273–278, Nov. 2002.

- [13]	E. S. Ford, “Does exercise reduce inflammation? Physical activity and C-reactive protein among U.S. adults,” Epidemiology, vol. 13, no. 5, pp. 561–568, Sep. 2002.

- [14] J. A. Sanford et al., “Molecular Transducers of Physical Activity Consortium (MoTrPAC): Mapping the Dynamic Responses to Exercise,” Cell, vol. 181, no. 7, pp. 1464–1474, Jun. 2020.

- [15] K. Jia, C. Cui, Y. Gao, Y. Zhou, and Q. Cui, “An analysis of aging-related genes derived from the Genotype-Tissue Expression project (GTEx),” Cell Death Discov, vol. 4, p. 26, Aug. 2018.

- [16] S. Schenk et al., “Physiological Adaptations to Progressive Endurance Exercise Training in Adult And Aged Rats: Insights from The Molecular Transducers of Physical Activity Consortium (MoTrPAC),” Function, doi: 10.1093/function/zqae014.

- [17] T. G. Consortium et al., “The Genotype-Tissue Expression (GTEx) pilot analysis: Multitissue gene regulation in humans,” Science, vol. 348, no. 6235, pp. 648–660, 2015.

- [18] E. Y. Chen et al., “Enrichr: interactive and collaborative HTML5 gene list enrichment analysis tool,” BMC Bioinformatics, vol. 14, p. 128, Apr. 2013.

- [19] OpenAI et al., “GPT-4 Technical Report,” arXiv [cs.CL], Mar. 15, 2023. [Online]. Available: http://arxiv.org/abs/2303.08774

- [20] Gene Ontology Consortium, “Gene Ontology Consortium: going forward,” Nucleic Acids Res., vol. 43, no. Database issue, pp. D1049–56, Jan. 2015.

- [21] M. Kutmon et al., “WikiPathways: capturing the full diversity of pathway knowledge,” Nucleic Acids Res., vol. 44, no. D1, pp. D488–94, Jan. 2016.

- [22] J. A. Blake, C. J. Bult, J. T. Eppig, J. A. Kadin, J. E. Richardson, and Mouse Genome Database Group, “The Mouse Genome Database genotypes::phenotypes,” Nucleic Acids Res., vol. 37, no. Database issue, pp. D712–9, Jan. 2009.

- [23] E. Sollis et al., “The NHGRI-EBI GWAS Catalog: knowledgebase and deposition resource,” Nucleic Acids Res., vol. 51, no. D1, pp. D977–D985, Jan. 2023.

- [24]	J. M. Memme, A. T. Erlich, G. Phukan, and D. A. Hood, “Exercise and mitochondrial health,” J. Physiol., vol. 599, no. 3, pp. 803–817, Feb. 2021.

- [25]	Y. Qiu et al., “Exercise sustains the hallmarks of health,” J Sport Health Sci, vol. 12, no. 1, pp. 8–35, Jan. 2023.

- [26]	R. P. Tracy and E. G. Bovill, “Thrombosis and cardiovascular risk in the elderly,” Arch. Pathol. Lab. Med., vol. 116, no. 12, pp. 1307–1312, Dec. 1992.

- [27]	M. S. El-Sayed, C. Sale, P. G. Jones, and M. Chester, “Blood hemostasis in exercise and training,” Med. Sci. Sports Exerc., vol. 32, no. 5, pp. 918–925, May 2000.

- [28]	A. A. Johnson and A. Stolzing, “The role of lipid metabolism in aging, lifespan regulation, and age-related disease,” Aging Cell, vol. 18, no. 6, p. e13048, Dec. 2019.

- [29]	Prospective Studies Collaboration et al., “Blood cholesterol and vascular mortality by age, sex, and blood pressure: a meta-analysis of individual data from 61 prospective studies with 55,000 vascular deaths,” Lancet, vol. 370, no. 9602, pp. 1829–1839, Dec. 2007.

- [30]	J. F. Horowitz and S. Klein, “Lipid metabolism during endurance exercise,” Am. J. Clin. Nutr., vol. 72, no. 2 Suppl, p. 558S–63S, Aug. 2000.

- [31]	P. V. Carapeto and C. Aguayo-Mazzucato, “Effects of exercise on cellular and tissue aging,” Aging , vol. 13, no. 10, pp. 14522–14543, May 2021.


